Change XY formatting on LCD

AnHardt opened the pull request:
According to MarlinFirmware/MarlinDev#123 negative values for XY at or below -100 are displaying incorrectly, dropping the first digit. Deltas can easily have XY values in this range. This PR adds a function to display floats/ints formatted like _123, -123, _-12, or __-1 as appropriate and applies it to the XY coordinates on Hitachi displays. It also moves the Z value to the right to be consistent with the XY formatting.

Originally by @thinkyhead (MarlinFirmware/MarlinDev#181)
with some minor improvements.

thinkyhead commented:
Thanks for patching this up, Andreas, I've been in another zone. It looks much better. This should be submitted to `MarlinDev/master` also.

One small style note. The "astyle" tool was keen to use this: `const float& x` instead of this: `const float &x` so that's what I've been using going forward, in case "astyle" should make another appearance.

AnHardt committed:
Change XY formatting on LCD

According to #123 negative values for XY at or below -100 are displaying incorrectly, dropping the first digit. Deltas can easily have XY values in this range. This PR adds a function to display floats/ints formatted like `_123`, `-123`, `_-12`, or `__-1` as appropriate and applies it to the XY coordinates on Hitachi displays. It also moves the Z value to the right to be consistent with the XY formatting.

AnHardt commented:
@thinkyhead 
What ever you want. Changed the formatting.

thinkyhead commented:
This looks about perfect now, and so does MarlinFirmware/MarlinDev#265. I have been using this code (on top of 1.1-RC2) on my own machine and it works well.

AnHardt commented:
@thinkyhead 
Somewhere here is a _`Merge-button`_.
